app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/RegistrationController.scala

```diff
@@ -16,34 +16,96 @@
+import uk.gov.hmrc.cgtpropertydisposalsfrontend.util.toFuture
 import uk.gov.hmrc.cgtpropertydisposalsfrontend.views
 import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 
+import scala.concurrent.{ExecutionContext, Future}
+
 @Singleton
 class RegistrationController @Inject()(
-                                        val authenticatedAction: AuthenticatedAction,
-                                        val sessionDataAction: SessionDataAction,
-                                        val sessionStore: SessionStore,
-                                        val errorHandler: ErrorHandler,
-                                        startRegistrationPage: views.html.registration.registration_start,
-                                        cc: MessagesControllerComponents
-                                      )(implicit viewConfig: ViewConfig)
+    val authenticatedAction: AuthenticatedAction,
+    val sessionDataAction: SessionDataAction,
+    val sessionStore: SessionStore,
+    val errorHandler: ErrorHandler,
+    startRegistrationPage: views.html.registration.registration_start,
+    selectEntityTypePage: views.html.registration.select_entity_type,
+    wrongGGAccountForTrustPage: views.html.wrong_gg_account_for_trust,
+    cc: MessagesControllerComponents
+  )(
+    implicit viewConfig: ViewConfig,
+    ec: ExecutionContext
+  )
   extends FrontendController(cc) with WithAuthAndSessionDataAction {
+  import RegistrationController._
 
-  def startRegistration(): Action[AnyContent] = authenticatedActionWithSessionData { implicit request =>
+  def carryOnIfValidUser(request: RequestWithSessionData[_])(f: IndividualWithInsufficientConfidenceLevel => Future[Result]): Future[Result] =
     request.sessionData.flatMap(_.subscriptionStatus) match {
-      case Some(IndividualWithInsufficientConfidenceLevel(Some(false), Some(HasSAUTR(None)), _, _)) =>
-        Ok(startRegistrationPage(routes.InsufficientConfidenceLevelController.doYouHaveAnSaUtr()))
-      case _ =>
-        SeeOther(routes.StartController.start().url)
+      case Some(u @ IndividualWithInsufficientConfidenceLevel(Some(false), Some(HasSAUTR(None)), _, _))
+        => f(u)
+      case _
+        =>  SeeOther(routes.StartController.start().url)
     }
+
+  def startRegistration(): Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    carryOnIfValidUser(request)(_ =>
+      Ok(startRegistrationPage(routes.InsufficientConfidenceLevelController.doYouHaveAnSaUtr()))
+    )
+  }
+
+  def selectEntityType():  Action[AnyContent] = authenticatedActionWithSessionData.async { implicit request =>
+    carryOnIfValidUser(request)(_ =>
+      Ok(selectEntityTypePage(RegistrationController.selectEntityTypeForm, routes.RegistrationController.startRegistration()))
+    )
+  }
+
+  def selectEntityTypeSubmit(): Action[AnyContent] =  authenticatedActionWithSessionData { implicit request =>
+    RegistrationController.selectEntityTypeForm.bindFromRequest().fold(
```

Comment:
we could make use of `carryOnIfValid` user here as well if we wanted to be totally anal about it